.github/workflows/ci.yml

```diff
@@ -1,48 +1,73 @@
 name: CI
 
-on: [push, pull_request]
+on:
+  push:
+  pull_request:
+  release:
+    types:
+      - published
 
 env:
-  PYTHON_VERSION: '3.10'
+  MINIMUM_PYTHON_VERSION: '3.6'
 
-jobs:
+concurrency:
+  group: ${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
 
+jobs:
   precommit:
     runs-on: ubuntu-latest
     steps:
-      - name: Install Python ${{ env.PYTHON_VERSION }}
+      - name: Install Python ${{ env.MINIMUM_PYTHON_VERSION }}
         uses: actions/setup-python@v2
         with:
-          python-version: ${{ env.PYTHON_VERSION }}
+          python-version: ${{ env.MINIMUM_PYTHON_VERSION }}
       - uses: actions/checkout@v2
       - name: Install pre-commit
         run: |
-          python -m pip install pre-commit
+          python -m pip install --upgrade pip
+          python -m pip install pre-commit virtualenv!=20.0.6
           pre-commit install
       - name: Run static code inspections
         run: pre-commit run --all-files
 
-  test:
-    needs: [precommit]
+  build:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         python: ["3.6", "3.7", "3.8", "3.9", "3.10"]
     steps:
       - uses: actions/checkout@v2
-      - name: Setup Python
+      - name: Set up Python ${{ matrix.python-version }}
```

Comment:
Possibly this should be `matrix.python`?

Reply:
Good catch! The usage seems OK, but this also happens on other repos where we test for 3.10 and use strings for the python version.

Reply:
You were right! Should be resolved now.